add/ properties를 yml파일로 교체(local, prod, 공통)

zz106603 opened the pull request:
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] ⚰️ Maintain (keep in good condition or in working)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
- .properties를 .yml로 전체 교체
- local, prod, 공통으로 사용
- active는 SecretKey를 통해 설정되도록 함(로컬은 env에서 local로 설정)

<!-- 이슈를 해결한 코드에 대해 설명해주세요. -->

### 📝 Checklist

<!-- 테스트 코드 구성이 가능하다면 포함해주세요. -->
<!-- 문서화가 따로 필요 없는 부분이라면 해당 항목을 삭제해주세요. -->

- [x] 🔴 Human eyes (no test)
- [ ] 🟡 swagger or Smoke test
- [ ] 🟢 unit test or CI test

### 📸 Log/Screenshot

<!-- 로그나 테스트 스크린샷이 있을 경우 첨부해주세요. -->

🌱 before

🏡 after

zz106603 committed:
# This is a combination of 16 commits.
# This is the 1st commit message:

add: properties를 yml파일로 교체(local, prod, 공통)

# This is the commit message #2:

feat: 오타 수정

# This is the commit message #3:

feat: build시 test코드 일시적으로 비활성화

# This is the commit message #4:

feat: yml 누락 내용 추가

# This is the commit message #5:

feat: 환경변수 적용 테스트

# This is the commit message #6:

feat: 환경변수 적용 확인 테스트

# This is the commit message #7:

feat: 환경변수 적용 테스트

# This is the commit message #8:

feat: 환경변수 적용 테스트

# This is the commit message #9:

feat: 환경변수 적용 테스트

# This is the commit message #10:

feat: 환경변수 적용 테스트

# This is the commit message #11:

feat: jwt 테스트를 위한 redirect 세팅

# This is the commit message #12:

feat: 쿠키값 전송 테스트

# This is the commit message #13:

feat: redirect 테스트

# This is the commit message #14:

feat: redirect 테스트

# This is the commit message #15:

feat: redirect 테스트

# This is the commit message #16:

feat: cors 추가(테스트 localhost:3000)

jsjouo249 commented on src/main/java/com/edio/EdioBackendApplication.java:
```diff
@@ -27,6 +27,15 @@ public static void main(String[] args) {
 
         System.setProperty("spring.profiles.active", dotenv.get("SPRING_PROFILES_ACTIVE"));
 
+        // 로그 출력
```
확인용이라면 지워도 되지 않을까요??

GuardJo replied:
옳소

jsjouo249 reviewed (approved):
LGTM!!

GuardJo commented on src/main/java/com/edio/common/security/SecurityConfig.java:
```diff
@@ -55,34 +60,36 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-//                          response.sendRedirect("http://localhost:3000/oauth2/redirect");
-                                })
+                        .userInfoEndpoint(userInfo -> userInfo.userService(customOAuth2UserService))
+                        .successHandler((request, response, authentication) -> {
```
이정도 사이즈면 SuccessHandler 별도 객체로 빼는건 어떨까요?

zz106603 replied:
> 이정도 사이즈면 SuccessHandler 별도 객체로 빼는건 어떨까요?

완료

GuardJo commented on src/main/java/com/edio/common/security/SecurityConfig.java:
```diff
@@ -110,17 +117,16 @@ public OAuth2AccessTokenResponseClient<OAuth2AuthorizationCodeGrantRequest> acce
+    @Bean
+    public CorsConfigurationSource corsConfigurationSource() {
+        CorsConfiguration configuration = new CorsConfiguration();
+        configuration.setAllowedOrigins(Arrays.asList("http://localhost:3000"));
```
이거 FE url이 local용이니 외부 변수로 빼두는게 좋을 듯

GuardJo replied:
아니면 와일드카드로 열거나

GuardJo commented on src/main/java/com/edio/common/security/SecurityConfig.java:
```diff
@@ -110,17 +117,16 @@ public OAuth2AccessTokenResponseClient<OAuth2AuthorizationCodeGrantRequest> acce
+    public CorsConfigurationSource corsConfigurationSource() {
+        CorsConfiguration configuration = new CorsConfiguration();
+        configuration.setAllowedOrigins(Arrays.asList("http://localhost:3000"));
+        configuration.setAllowedMethods(Arrays.asList("GET", "POST", "PUT", "DELETE", "OPTIONS"));
```
PATCH도 넣어주세요

zz106603 committed:
feat: fe와 jwt토큰 전송 테스트

zz106603 committed:
feat: oauth successHanler 분리 및 불필요 로그 제거

feat: build시 test되도록 수정

feat: Test 코드 env환경변수 설정

feat: deploy.yml 수정(테스트 진행)

feat: clean build 적용

feat: build시에 cache 무시

feat: build 에러 추적

feat: build시 로그 추가

feat: deploy.sh 복구

feat: beforeInstall 설정 추가

feat: cleanup 추가

feat: clenup 내용 수정

feat: dofilter 로그 추가

feat: httpOnly false로 설정

feat: security 로그 추가

feat: security 로그 추가

feat: security 로그 추가

feat: 빈 등록 확인 로그 추가

feat: log 레벨 수정

feat: 로그 레벨 수정

feat: samsite 설정

feat: cookie세팅 수정

feat: 쿠키 수정

feat: secure 설정

feat: samesite 설정

feat: webConfig 추가

feat: 쿠키 설정

feat: 쿠키 세팅 원상 복구

zz106603 committed:
feat: EC2 도메인 변경

GuardJo reviewed (approved):
👍
